build.sbt

```diff
@@ -302,12 +312,24 @@ lazy val java = project
 
 lazy val semconv = crossProject(JVMPlatform, JSPlatform, NativePlatform)
   .crossType(CrossType.Pure)
+  .enablePlugins(BuildInfoPlugin)
   .in(file("semconv"))
   .dependsOn(`core-common`)
   .settings(
     name := "otel4s-semconv",
     startYear := Some(2023),
+    // We use opentelemetry-semconv dependency to track releases of the OpenTelemetry semantic convention spec
+    libraryDependencies += "io.opentelemetry.semconv" % "opentelemetry-semconv" % OpenTelemetrySemConvVersion % "compile-internal" intransitive (),
+    buildInfoPackage := "org.typelevel.otel4s.semconv",
+    buildInfoOptions += sbtbuildinfo.BuildInfoOption.PackagePrivate,
+    buildInfoKeys := Seq[BuildInfoKey](
+      "openTelemetrySemanticConventionsVersion" -> OpenTelemetrySemConvVersion
+    )
   )
+  .jsSettings(
+    scalacOptions += "-Wconf:cat=deprecation:s"
```

Comment:
ScalaJS build fails without this flag :( 

https://github.com/typelevel/otel4s/actions/runs/6547484874/job/17780063997?pr=336

Reply:
Can we use `@nowarn` instead of the flag?

Reply:
Good point, `@nowarn` fits better.